### ISSUE TITLE
Most of 2018 Firefox nightly build data is missing from Buildhub v2

### ISSUE THREAD

**Issue by mars-f** [mozilla-releng/buildhub2#473]
I'm trying to construct a search query against 2018 Firefox nightly builds in Buildhub v2 and I am seeing an order of magnitude fewer results than are returned from Buildhub v1.  The results in https://buildhub.moz.tools only go back to ~Nov 27th 2018.  The rest of 2018 is missing. There are results for 2017 and back.

This query returns 753 results:
https://mozilla-services.github.io/buildhub/?q=2018&platform[0]=linux-x86_64&channel[0]=nightly&locale[0]=en-US&products[0]=firefox

This query returns only 73:
https://buildhub.moz.tools/?q=2018&platform[0]=linux-x86_64&channel[0]=nightly&locale[0]=en-US&products[0]=firefox

I expect https://buildhub.moz.tools to return the same set of results as https://mozilla-services.github.io/.

**Comment by peterbe** [comment]
It seems, that something might have done wrong in the migration. 

First of all, @autrilla and I naively jumped in to see if it was a problem with Elasticsearch. Short story; no, it seems Elasticsearch is matching its true source which is the PostgreSQL database. 

We bypassed ES and sent some raw queries to Postgres to see roughly what's in there, filtered by buildID:

|       | Stage | Prod |
|-------|-------|------|
| 2016% | 118873      | 128964     |
| 2017% | 332333     |  311388    |
| 2018% | 1263   | 1138     |
| 2019% | 1178   | 1178     |

And specifically for `2018...`

|       | Stage | Prod |
|-------|-------|------|
| 201810% | 0   | 0     |
| 201811% | 273   | 148     |
| 201812% | 990   | 990     |
| last 2 summed| 1263 | 1138 |

Remember, when we migrated data from Buildhub1 to Buildhub2 we got all locales. But from the moment after the migration and onwards Buildhub2 will only pick up en-US builds. 

What is curious is that we only appear to have things in 2018 for November and December. 

Basically, it seems we have *nothing* from the migration for those with buildID `2018...`.

**Comment by peterbe** [comment]
I'm going to attempt to do the migration all over again and see what happens. Perhaps we skipped too many records.

**Comment by peterbe** [comment]
Note-to-self; 

I got a snapshot of Buildhub1's full (Prod) Kinto database and ran it locally like this:

```
./manage.py kinto-database-migration --skip-invalid
...
In total, skipped 1,228,833 invalid records.
In total, processed 1,703,455 valid records.
In total, inserted 0 valid records.
The whole migration took 3.9 minutes.
```

I appreciate that there's a lot of "junk" in there that rightfully shouldn't get migrated. But I think we now need to investigate those 1,228,833 rows in some fashion.

**Comment by peterbe** [comment]
I found it! The migration was skipping a LOT of records because of something simple. I wiped by local Postgres and ran the `./manage.py kinto-database-migration` with fresh pg dump from Buildhub1 (thank you @bqbn).

A lot of build records from Kinto look like this:

```python
{'build': {'as': 'ml64.exe',
           'cc': 'z:/build/build/src/vs2015u3/VC/bin/amd64/cl.exe',
           'cxx': 'z:/build/build/src/vs2015u3/VC/bin/amd64/cl.exe',
           'date': '2017-09-18T22:00:54Z',
           'host': 'x86_64-pc-mingw32',
           'id': '20170918220054',
           'target': 'x86_64-pc-mingw32'},
 'download': {'date': '2017-09-19T04:11:33Z',
              'mimetype': 'application/msdos-windows',
              'size': 40633720,
              'url': 'https://archive.mozilla.org/pub/firefox/nightly/2017/09/2017-09-18-22-00-54-mozilla-central-l10n/firefox-57.0a1.es-MX.win64.installer.exe'},
 'schema': 1519839534729,
 'source': {'product': 'firefox',
            'repository': 'https://hg.mozilla.org/mozilla-central',
            'revision': '42151fcd6cfc216d147730d0f2c6a2acd52d22fd',
            'tree': 'mozilla-central'},
 'target': {'channel': 'nightly',
            'locale': 'es-MX',
            'os': 'win',
            'platform': 'win64',
            'version': '57.0a1'}}
```
This one fails JSON Schema validation because it has a extra key that is not expected to be there:
```
Additional properties are not allowed ('schema' was unexpected)
```

So I analyzed how many of these have that:
```
HAS SCHEMA: 1,228,833
NOT SCHEMA: 474,622
In total, skipped 1,228,833 invalid records.
In total, processed 1,703,455 valid records.
In total, inserted 0 valid records.
The whole migration took 3.7 minutes.
```
I.e. 72% of Kinto records have that one!

If I add a simple `build.pop('schema', None)` and run again I get this:

```
In total, skipped 0 invalid records.
In total, processed 1,703,455 valid records.
In total, inserted 1,228,833 valid records.
The whole migration took 14.8 minutes.
```

Run it one more time and this happens:

```
In total, skipped 0 invalid records.
In total, processed 1,703,455 valid records.
In total, inserted 0 valid records.
```

Now, after I've run it I get the same numbers as Buildhub1.

**Comment by peterbe** [comment]
@autrilla We'll land https://github.com/mozilla/buildhub2/pull/474 and make it go to stage. Then I'll ask you to do the kinto database migration from a fresh'ish pg dump from Buildhub1 and then we'll run that `reindex-elasticsearch` and if I'm right all shall be well with Buildhub2 and historical data. 

One caveat; when we do this we are going to get a bunch of non en-US builds into our more pristine Buildhub2. We can consider to delete them all at some point but that's for another day to decide.

**Comment by peterbe** [comment]
@autrilla made the migration (with the new code from last Friday) on Stage and now it matches Buildhub1 Prod.

Buildhub1 Prod:
<img width="779" alt="screen shot 2019-02-04 at 12 38 58 pm" src="https://user-images.githubusercontent.com/26739/52226058-f2225180-2879-11e9-82ab-7f41282a046f.png">

Buildhub2 Stage:
<img width="837" alt="screen shot 2019-02-04 at 12 37 17 pm" src="https://user-images.githubusercontent.com/26739/52226081-fbabb980-2879-11e9-892e-6badcf5b3f01.png">


@autrilla When time allows, do the Prod push and the whole migration thing you did on Stage please.

**Comment by autrilla** [comment]
@peterbe deployed to prod and did the whole migration thing.

```
In total, skipped 0 invalid records.
In total, processed 1,711,126 valid records.
In total, inserted 1,236,504 valid records.
The whole migration took 13.7 minutes.
```

ES reindex also ran successfully.
